### PR TITLE
bsunanda:Run2-hcx60 Provide hexgonal geometry for HGCal

### PR DESCRIPTION
**Pull request by bsunanda**
This will give simulation as well as reconstruction geometry for HGCal (square as well as hexgonal cell). The hook for DB will require some work from Yana. 
@kpedro88 @lgray Please take a look

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by lgray** [comment]
Hi Sunanda, is the default geometry still square in this setup?

**Comment by lgray** [comment]
How do I import this geometry into a CMSSW config? Is there an example config?
Don't see one in the files for this commit.

**Comment by bsunanda** [comment]
I did not provide a new config because David always picks on that. I tested this using a test python file in Geometry/HGCalCommonData/python/testHGCXML_cfi.py. I can make a second such in Geometry/CMSCommonData and Configuration/Geometry. But I shall put them in a separate PR.
________________________________

**Comment by lgray** [comment]
If you can provide a V7 config privately I'll be able to check that things
start to make sense. Just send me something by email.

-L

File: CondFormats/GeometryObjects/interface/HGCalParameters.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    int           lay;
+    float         bl, tl, h, dz, alpha, cellSize;
+  };
+
+  struct hgtrform {
+    int                zp, lay, sec, subsec;
+    bool               used;
+    CLHEP::Hep3Vector  h3v;
+    CLHEP::HepRotation hr;
+  };
```
Comment (ianna) [inline CondFormats/GeometryObjects/interface/HGCalParameters.h]: @bsunanda - this object should define what is stored to/retrieved from DB. I think, using vectors of doubles is better.

File: CondFormats/GeometryObjects/interface/HGCalParameters.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#ifndef CondFormats_GeometryObjects_HGCalParameters_h
+#define CondFormats_GeometryObjects_HGCalParameters_h
+
+#include "CondFormats/Serialization/interface/Serializable.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include<string>
+#include<vector>
+#include<iostream>
+#include<CLHEP/Geometry/Transform3D.h>
```
Comment (ianna) [inline CondFormats/GeometryObjects/interface/HGCalParameters.h]: @bsunanda - please, add a space in the lines above. I think, using clhep here is not a good idea.

File: CondFormats/GeometryObjects/src/classes_def.xml [hunk 3]
```diff
@@ -89,6 +89,43 @@
...
+   <field name="boundR_"          mapping="blob"/>  
+   <field name="waferR_"          mapping="blob"/>  
+   <field name="mode_"            mapping="blob"/>  
+ </class>
+ <class name="HGCalParameters::hgtrap"/>
+ <class name="std::vector<HGCalParameters::hgtrap>"/>
+ <class name="HGCalParameters::hgtrform"/>
+ <class name="std::vector<HGCalParameters::hgtrform>"/>
+ -->
+
```
Comment (ianna) [inline CondFormats/GeometryObjects/src/classes_def.xml]: @bsunanda - why is it commented out?

File: Geometry/HGCalCommonData/data/v7/hgcalCons.xml [hunk 4]
```diff
@@ -2,253 +2,384 @@
...
-    <Parameter name="GroupingZ"  value="26"/>
-    <Parameter name="GroupingZ"  value="26"/>
-    <Parameter name="GroupingZ"  value="-1"/>
-    <Parameter name="GroupingZ"  value="27"/>
-    <Parameter name="GroupingZ"  value="27"/>
-    <Parameter name="GroupingZ"  value="-1"/>
-    <Parameter name="GroupingZ"  value="28"/>
-    <Parameter name="GroupingZ"  value="28"/>
-    <Parameter name="GroupingZ"  value="-1"/>
+    <Parameter name="GroupingZFine"  value="1"/>
```
Comment (ianna) [inline Geometry/HGCalCommonData/data/v7/hgcalCons.xml]: @bsunanda - please, use Vector here.

File: Geometry/HGCalCommonData/data/v7/hgcalCons.xml [hunk 5]
```diff
@@ -2,253 +2,384 @@
...
-    <Parameter name="GroupingZ"  value="10"/>
-    <Parameter name="GroupingZ"  value="10"/>
-    <Parameter name="GroupingZ"  value="10"/>
-    <Parameter name="GroupingZ"  value="11"/>
-    <Parameter name="GroupingZ"  value="11"/>
-    <Parameter name="GroupingZ"  value="11"/>
-    <Parameter name="GroupingZ"  value="12"/>
-    <Parameter name="GroupingZ"  value="12"/>
-    <Parameter name="GroupingZ"  value="12"/>
+    <Parameter name="GroupingZFine"  value="1"/>
```
Comment (ianna) [inline Geometry/HGCalCommonData/data/v7/hgcalCons.xml]: @bsunanda - please, use Vector.

File: Geometry/HGCalGeometry/plugins/HGCalGeometryESProducer.cc [hunk 6]
```diff
@@ -30,7 +30,7 @@
 #include "Geometry/HGCalGeometry/interface/HGCalGeometry.h"
 #include "Geometry/HGCalGeometry/interface/HGCalGeometryLoader.h"
 
-//#define DebugLog
+#define DebugLog
```
Comment (ianna) [inline Geometry/HGCalGeometry/plugins/HGCalGeometryESProducer.cc]: @bsunanda - it's better to use LogDebug which can be switched on/off without editing the code. In any case, this line should be commented out when it's submitted for a release.

File: SimDataFormats/CaloTest/src/HGCalTestNumbering.cc [hunk 7]
```diff
@@ -1,22 +1,12 @@
 #include "SimDataFormats/CaloTest/interface/HGCalTestNumbering.h"
 #include <iostream>
 
-//#define DebugLog
+#define DebugLog
```
Comment (ianna) [inline SimDataFormats/CaloTest/src/HGCalTestNumbering.cc]: @bsunanda - please, comment it out.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna Took care of most of your comments - except use of vector. The same array name is used in different context.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test
@lgray @davidlange6 The tests for HGCal are failing due to global tag issue (the appropriate thing for RPC cluster is not found) -clearly this has nothing to do with the changed codes here.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/12845	 9ef390e
- minimal changes in RECO: only geometry include file changed in RecoLocalCalo/HGCalRecProducers/test/testHGCalRecoLocal_cfg.py 
- jenkins tests pass and comparisons with baseline show no differences (sanity); the unit tests in RecoLocalCalo/HGCalRecProducers run to completion.

**Comment by ianna** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/12845	 d51b973
- since 9ef390e, the changes are only in Geometry/HGCalCommonData and do not affect reco
- jenkins tests pass and comparisons show no differences

**Comment by ianna** [comment]
@bsunanda - please, close this PR. #12862 should replace it. Thanks.

**Comment by ianna** [comment]
-1

**Pull request closed without merging**